react/ContactsList/ContactsList.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import { sortLastNameFirst, buildLastNameFirst } from './'
+import ContactRow from './ContactRow'
+import ContactHeaderRow from './ContactHeaderRow'
+import styles from './styles.styl'
+
+const ContactsList = props => {
+  const { contacts, className, onItemClick, ...rest } = props
+
+  const sortedContacts = [...contacts].sort(sortLastNameFirst)
+  const categorizedContacts = sortedContacts.reduce((acc, contact) => {
+    const name = buildLastNameFirst(contact)
+    const header = name[0] || 'EMPTY'
+    acc[header] = acc[header] || []
+    acc[header].push(contact)
+    return acc
+  }, {})
```

Comment:
I think we should avoid sorting and building sorted contacts inside the render as it can be fairly expensive (we do not want to explode our 16ms budget ;))

Doing this via reselect would be better IMHO.

Reply:
@drazik can you add a TODO with @ptbrowne's comment ? :)